### PR TITLE
Net6.0 Minimal API projects without Startup, issue #3794

### PR DESCRIPTION
**Pull request by bsurprised**
Fixes missing controllers in the specification when generating Net6.0 Minimal API projects using CLI/MSBuild

### CODE REVIEW

File: src/NSwag.Commands/HostApplication.cs [hunk 1]
```diff
@@ -140,7 +140,13 @@ void OnEntryPointExit(Exception exception)
             try
             {
                 // Get the IServiceProvider from the host
+#if NET6_0
```
Comment (maxkoshevoi) [inline src/NSwag.Commands/HostApplication.cs:143]: `NET6_0_OR_GEATER` or else it will be broken in NET 7 again

Comment (bsurprised) [reply src/NSwag.Commands/HostApplication.cs:143]: Good idea. 
From my understanding, NSwag style has been code tailored to each dotnet version and I made it NET6 specific. DotNet has had many breaking changes in recent versions, so I do hope this works for NET7 as well. :-)

**Comment by vpetrusevici** [comment]
Hello @RicoSuter 
Can you say please when this fix will be released?

**Comment by lahma** [comment]
@vpetrusevici There's now a new release on NuGet

**Comment by vip32** [comment]
just tried it, works fine :) However when also setting up API versioning (Microsoft.AspNetCore.Mvc.Versioning.ApiExplorer) with the following setup in Program.cs. Note that the versioning is not really used in the minimal endpoints or controller.

`builder.Services.AddApiVersioning(options => ConfigureApiVersioning(options));
builder.Services.AddVersionedApiExplorer(options => options.SubstituteApiVersionInUrl = true);`

the minimal endpoints are *not* present in the swagger, the controller operations are present.

====================================================================
this is the whole sample (NSwag.Sample.NET60Minimal) is used:

using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.Mvc.Versioning;
using Microsoft.Extensions.DependencyInjection;
using System;

var builder = WebApplication.CreateBuilder(args);

// Optional: Use controllers
builder.Services.AddControllers();

builder.Services.AddApiVersioning(options => ConfigureApiVersioning(options));
builder.Services.AddVersionedApiExplorer(options => options.SubstituteApiVersionInUrl = true);

builder.Services.AddEndpointsApiExplorer();
builder.Services.AddOpenApiDocument(settings =>
{
    settings.Title = "Minimal API";
    settings.Version = "v1";
});

var app = builder.Build();
app.UseDeveloperExceptionPage();

app.UseOpenApi();
app.UseSwaggerUi3();

app.MapGet("/", (Func<string>)(() => "Hello World!"))
    .WithTags("General123");

app.MapGet("/sum/{a}/{b}", (Func<int, int, int>)((a, b) => a + b))
    .WithName("CalculateSum")
    .WithTags("Calculator");

// Optional: Use controllers
app.UseRouting();
app.MapControllers();

app.Run();

void ConfigureApiVersioning(ApiVersioningOptions options)
{
    options.DefaultApiVersion = new ApiVersion(1, 0);
    options.AssumeDefaultVersionWhenUnspecified = true;
    options.ReportApiVersions = true;
}

// Optional: Use controllers
[ApiController]
[Route("api/examples")]
public class ExampleController : ControllerBase
{
	[HttpGet]
	public IActionResult Get()
	{
		return Ok("Examples");
	}
}

**Comment by RicoSuter** [comment]
> However when also setting up API versioning (Microsoft.AspNetCore.Mvc.Versioning.ApiExplorer) with the following line in Program.cs. Note that the versioning is not really used in the minimal endpoints or controller.

Seems that for some reason the minimal operations are not reported anymore by API Explorer... needs investigating.

**Comment by RicoSuter** [comment]
Is this still a problem with v13.15.10?

**Comment by vip32** [comment]
yes, just tried again with 13.15.10. `AddVersionedApiExplorer` still causes minimal endpoints to be *not* present in the generated swagger json.

**Pull request merged**